agent/lib_aws_sdk_php.c

```diff
@@ -0,0 +1,183 @@
+
+#define PHP_PACKAGE_NAME "aws/aws-sdk-php"
+#define PHP_AWS_SDK_SERVICE_NAME_METRIC_PREFIX \
+  "Supportability/PHP/AWS/Services/"
+
+/*
+ * In a normal course of events, the following line will always work
+ * zend_eval_string("Aws\\Sdk::VERSION;", &retval, "Get AWS Version")
+ * By the time we have detected the existence of the aws-sdk-php and with
+ * default composer profject settings, it callable even from
+ * nr_aws_sdk_php_enable which will automatically load the class if it isn't
+ * loaded yet and then evaluate the string. However, in the rare case that files
+ * are not loaded via autoloader and/or have non-default composer classload
+ * settings, if the class is not found, PHP 8.2+ will generate a fatal
+ * unrecoverable uncatchable error error whenever it cannot find a class. While
+ * calling this from nr_aws_sdk_php_enable would have been great and would allow
+ * the sdk version value to be set only once, to avoid the very unlikely but not
+ * impossible fatal error, this will be called from the
+ * "Aws\\ClientResolver::_apply_user_agent" wrapper which GUARANTEES that
+ * aws/sdk exists and is already loaded.
+ *
+ *
+ * Additionally given that aws-sdk-php is currently detected from the
+ * AwsClient.php file, this method will always be called when a client is
+ * created unlike Sdk::construct which doesn't show with PHP 8.2+.
+ *
+ * Using Aws/Sdk::__construct for version is currently nonviable as it is
+ * unreliable as a version determiner.
+ * Having separate functionality to extract from Aws/Sdk::__construct
+ * is both not required and is redundant and causes additional overhead and
+ * so only one function is needed to extract version.
+ *
+ * Aws\\ClientResolver::_apply_user_agent a reliable function as it is
+ * always called on client initialization since it is key to populating
+ * the request headers, and it loads Sdk by default.
+ *
+ * Concerns about future/past proofing to the checking prioritized the following
+ * implementation vs using the eval method.
+ */
+void nr_lib_aws_sdk_php_handle_version() {
+  zval* zval_version = NULL;
+  zend_class_entry* class_entry = NULL;
+  char* version = NULL;
+
+  class_entry = nr_php_find_class("aws\\sdk");
+  if (NULL != class_entry) {
+    zval_version = nr_php_get_class_constant(class_entry, "VERSION");
+
+    if (nr_php_is_zval_non_empty_string(zval_version)) {
+      version = Z_STRVAL_P(zval_version);
+    }
+  }
+  if (NRINI(vulnerability_management_package_detection_enabled)) {
+    /* Add php package to transaction */
+    nr_txn_add_php_package(NRPRG(txn), PHP_PACKAGE_NAME, version);
+  }
+  nr_fw_support_add_package_supportability_metric(NRPRG(txn), PHP_PACKAGE_NAME,
+                                                  version);
+  nr_php_zval_free(&zval_version);
+}
+
+void nr_lib_aws_sdk_php_add_supportability_service_metric(
+    const char* service_name) {
+  /* total MAX metric name length per agent-specs */
+  int MAX_METRIC_NAME_LEN = 256;
```

Comment:
This really should be a macro to avoid allocation on stack:
```suggestion
#define MAX_METRIC_NAME_LEN 256
```

Reply:
please see: [842d437](https://github.com/newrelic/newrelic-php-agent/pull/932/commits/842d4377f3be22d2feca2dd1f32e429ff4bd6a40)